### DISCUSSION TITLE
Unable to send PDF file from Edge Functions

### DISCUSSION THREAD

**Discussion by monbgul** [supabase/supabase#14958] [Questions]
EDGE FUNCTION:
![image](https://github.com/supabase/supabase/assets/104251469/d1bf82d5-078e-4df4-8410-eb53d1194d65)

ReactJS:
![image](https://github.com/supabase/supabase/assets/104251469/ea49d7b7-1c95-4a82-8acd-fce5fc53bc73)

In the above response, data is null and error is
![image](https://github.com/supabase/supabase/assets/104251469/1704a643-1eac-494a-8b46-6ccb0e463960)

Is possible to send PDF files from Edge Functions? If so, what is the issue in the above code?

**Comment by ilbertt** [comment]
@monbgul what library are you using for the `createPdf()` function? I'm also trying to generate PDFs in a Supabase Edge Function, but I can't find any library that works in Deno.

**Reply by filippobrigati** [reply]
Hi, @ilbertt you can use PDF-LIB

`import { PDFDocument } from 'https://cdn.pika.dev/pdf-lib@^1.7.0';`

[Here](https://pdf-lib.js.org/) the library docs
